website/static/js/tests/registrationRetraction.test.js

```diff
@@ -0,0 +1,127 @@
+            var redirectUrl = '/project/abdef/';
+            var response = {redirectUrl: redirectUrl};
+            var endpoints = [
+                {
+                    method: 'POST',
+                    url: submitUrl,
+                    response: response
+                },
+                {
+                    method: 'POST',
+                    url: invalidConfirmationText,
+                    response: {status: 500}
+                }
+
+            ];
+            var server;
+
+
+            before(() => {
+                server = utils.createServer(sinon, endpoints);
+            });
+
+            after(() => {
+                server.restore();
+            });
+
+            beforeEach(() => {
+                vm = new registrationRetraction.ViewModel(submitUrl, registrationTitle);
+            });
+
+            it('non-matching registration title is invalid', () => {
+                vm.confirmationText(invalidConfirmationText);
+                assert.isFalse(vm.confirmationText.isValid());
+            });
+
+            it('matching registration title is valid', () => {
+                 vm.confirmationText(registrationTitle);
+                assert.isTrue(vm.confirmationText.isValid());
+            });
+
+            describe('submit', () => {
+                var growlSpy;
+                var postSpy;
+
+                beforeEach(() => {
+                    growlSpy = new sinon.stub($osf, 'growl');
+                    postSpy = new sinon.spy($osf, 'postJSON');
+                });
+
+                afterEach(() => {
+                    growlSpy.restore();
+                    postSpy.restore();
+                });
+
+                it('calls growl if invalid confirmation text submitted', () => {
+                    vm.confirmationText(invalidConfirmationText);
+                    vm.submit();
+                    assert.calledOnce(growlSpy);
+                    assert.notCalled(postSpy);
+                });
+                it('calls growl if justification is too long', () => {
+                    vm.confirmationText(registrationTitle);
+                    vm.justification(invalidJustification);
+                    vm.submit();
+                    assert.calledOnce(growlSpy);
+                    assert.notCalled(postSpy);
+                });
+                it('submits successfully with valid confirmation text', (done) => {
+                    var onSubmitSuccessStub = new sinon.stub(vm, 'onSubmitSuccess');
+
+                    vm.confirmationText(registrationTitle);
+                    vm.submit().done(() => {
+                        assert.equal(response, redirectUrl);
+                        assert.called(onSubmitSuccessStub);
+                        assert.called(postSpy);
+                        assert.notCalled(growlSpy);
+                    });
+
+                    onSubmitSuccessStub.restore();
+                    done();
```

Comment:
`done()` should be called within the `done()` callback on ~L95 to ensure that the assertions are made.